modules/gapi/src/compiler/passes/intrin.cpp

```diff
@@ -238,10 +240,10 @@ void apply(cv::gimpl::GModel::Graph &g) {
         if (g.metadata(nh).get<NodeType>().t == NodeType::OP) {
             const auto &op = g.metadata(nh).get<Op>();
             if (op.k.name == cv::gapi::streaming::detail::GDesync::id()) {
-                auto index = g.metadata(nh).get<DesyncPath>().index;
+                auto cfg = g.metadata(nh).get<DesyncPath>().cfg;
                 auto new_links = drop(g, nh);
                 for (auto &&eh : new_links) {
-                    g.metadata(eh).set(DesyncEdge{index});
+                    g.metadata(eh).set(DesyncEdge{cfg});
```

Comment:
This and all above changes look unnecessary. Do you really need the desync's `drop_last` option all the way down??